src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs

```diff
@@ -656,7 +656,7 @@ public override async Task<IList<OrchestrationState>> GetStateAsync(IEnumerable<
             // to only be a small handful of parallel requests, so keeping the code simple until the storage refactor adds global throttling.
             var instanceQueries = instanceIds.Select(instance => this.GetStateAsync(instance, allExecutions: true, fetchInput: false));
             IEnumerable<IList<OrchestrationState>> instanceQueriesResults = await Task.WhenAll(instanceQueries);
-            return instanceQueriesResults.SelectMany(result => result).ToList();
+            return instanceQueriesResults.SelectMany(result => result).Where(x => x.OrchestrationInstance.InstanceId != null).ToList();
```

Comment:
I don't have full context on this PR, but isn't it possible that `x` could be null? If so, do we need to check for that too? Looking at the overloads of `GetStateAsync(string, bool, bool)`, null values for `OrchestrationState` seem possible.
```suggestion
            return instanceQueriesResults.SelectMany(result => result).Where(x => x?.OrchestrationInstance.InstanceId != null).ToList();
```

Reply:
The method that GetStateAsync calls,  FetchInstanceStatusInternalAsync will never return null. It's currently coded in an incorrect way. Currently in code, we return null if the DynamicEntityTable result is null. However, this will never be null even when there is no result returned. This means that FetchInstanceStatusInternalAsync will always return an InstanceStatus with an OrchestrationState. The OrchestrationState will just have null values.

This is something I am going to open an issue to address, as this code should be fixed to avoid any potential NullRefs in the future when using OrchestrationState. However, that will require a bit more code change and testing so this PR is intended to fix the NullRef on line 227 of OrchestrationSessionManager only.